### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by bdvx**
[Task 6](https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task6-sqs-sns/task.md)

- [ ] 1 - File serverless.yml contains configuration for catalogBatchProcess function
- [ ] 2 - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [ ] 3 - File serverless.yml contains configuration for SQS catalogItemsQueue
- [ ] 4 - File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription
- [ ] +1 - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

### CODE REVIEW

**Review by ilya-moskovtsev — COMMENTED** [review]
Done nothing yet?
0 out of 6 points?

**Comment by ilya-moskovtsev** [comment]
> Done nothing yet?
> 0 out of 6 points?

4/6 Will +1 for tests. Will +1 for Filter Policy.

**Comment by bdvx** [comment]
> Done nothing yet?
> 0 out of 6 points?

@ilya-moskovtsev Can you check again please? Had some tech issues and couldn't push the code

**Comment by bdvx** [comment]
@ilya-moskovtsev Can you look once again? Added filter policy

**Comment by ilya-moskovtsev** [comment]
> @ilya-moskovtsev Can you look once again? Added filter policy

Guess you did not intend to lose messages when price == 15000
5/6 Will +1 for tests.

File: product-service/src/catalogBatchProcess.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+export const catalogBatchProcess = async (event) => {
+    console.log('catalogBatchProcess lambda called with event: ', event);
+
+    try {
+        const { REGION: region } = process.env;
+        const sns = new AWS.SNS({ region });
+        const items = event.Records.map(({ body }) => JSON.parse(body));
+        // will add data validation later, as it's not a part of the task but good to have
+        items.forEach(async (item) => { await createProduct(item) });
```
Comment (svetlanagorchanina) [inline product-service/src/catalogBatchProcess.js:13]: Should not there be an import for `createProduct`?

**Pull request merged**